client/src/components/TOTP/tests/Register-test.js

```diff
@@ -1,193 +1,118 @@
-      const wrapper = shallow(
-        <Register
-          onBack={onBackMock}
-          onCompleteRegistration={onCompleteRegistrationMock}
-          method={mockMethod}
-          code="FOO123"
-          uri="example"
-          TOTPVerifyComponent={TOTPVerifyComponent}
-        />
-      );
-
-      wrapper.find('.mfa-action-list .btn').at(1).simulate('click');
-      expect(onBackMock.mock.calls.length).toBe(1);
-    });
-
-    it('goes to the input validation screen when clicking "Next" on the QR code screen', () => {
-      const wrapper = shallow(
-        <Register
-          onBack={onBackMock}
-          onCompleteRegistration={onCompleteRegistrationMock}
-          method={mockMethod}
-          code="FOO123"
-          uri="example"
-          TOTPVerifyComponent={TOTPVerifyComponent}
-        />
-      );
+function makeProps(obj = {}) {
+  return {
+    onBack: () => null,
+    onCompleteRegistration: () => null,
+    method: {
+      urlSegment: 'totp',
+      name: 'TOTP',
+      description: 'Register using TOTP',
+      supportLink: 'https://google.com',
+      component: 'TOTPRegister',
+    },
+    code: 'FOO123',
+    uri: 'example',
+    TOTPVerifyComponent: ({ moreOptionsControl }) => (
+      <div data-testid="totpverifycomponent">
+        {moreOptionsControl}
+      </div>
+    ),
+    ...obj
+  };
+}
+
+test('Register handleBack() calls the onBack prop', async () => {
+  const onBack = jest.fn();
+  render(
+    <Register {...makeProps({
+      onBack
+    })}
+    />
+  );
+  const back = await screen.findByText('Back');
+  fireEvent.click(back);
+  expect(onBack).toHaveBeenCalled();
+});
 
-      wrapper.find('.mfa-action-list .btn').first().simulate('click');
-      expect(wrapper.find(TOTPVerifyComponent)).toHaveLength(1);
-    });
-  });
+test('Register renderErrorScreen renders the providded errors', async () => {
+  render(
+    <Register {...makeProps({
+      errors: ['Something went wrong', 'I am a unit test'],
+    })}
+    />
+  );
+  const el = await screen.findByText('Something went wrong, I am a unit test');
+  expect(el.classList).toContain('mfa-totp__errors');
+});
 
-  describe('renderScanCodeScreen()', () => {
-    it('renders a QR code', () => {
-      const wrapper = shallow(
-        <Register
-          onBack={onBackMock}
-          onCompleteRegistration={onCompleteRegistrationMock}
-          method={mockMethod}
-          code="FOO123"
-          uri="example"
-          TOTPVerifyComponent={TOTPVerifyComponent}
-        />
-      );
+test('Register handleBackToScan() clears errors when clicking on the back button', async () => {
+  render(
+    <Register {...makeProps({
+      error: 'Something went wrong'
+    })}
+    />
+  );
+  const back = await screen.findByText('Back');
+  fireEvent.click(back);
+  const el = await screen.findByText('How to use authenticator apps.');
+  expect(el).not.toBeNull();
+});
```

Comment:
This isn't actually checking if an error has been cleared - we should validate that an element matching `.mfa-totp__errors` isn't there.

Reply:
Updated